internal/codegen/golang/opts/options.go

```diff
@@ -52,6 +53,30 @@ func ParseOpts(req *plugin.CodeGenRequest) (*Options, error) {
 		return options, fmt.Errorf("unmarshalling options: %w", err)
 	}
 
+	for i := range options.QuerySetOverrides {
+		if err := options.QuerySetOverrides[i].Parse(); err != nil {
+			return options, err
+		}
+
+		// construct a "plugin"-style override to make the next loop simpler
+		override := pluginOverride(req.Catalog.DefaultSchema, options.QuerySetOverrides[i])
+
+		// in sqlc config.Combine() the "package"-level configs are appended to global configs,
+		// so we mimic that behavior here
+		req.Settings.Overrides = append(req.Settings.Overrides, override)
+	}
+
+	for _, override := range req.Settings.Overrides {
+		var goType ParsedGoType
+		if err := json.Unmarshal(override.CodeType, &goType); err != nil {
+			return options, err
+		}
+		options.Overrides = append(options.Overrides, GoOverride{
+			override,
+			&goType,
+		})
+	}
```

Comment:
This feels backwards to me. We shouldn't need to serialize the query set overrides back to JSON only to parse them again a few lines later. Can't we iterate over req.Settings.Overrides and then iterate over the QuerySetOverrides?

Reply:
Yeah we can, I think this is a remnant of a different approach I abandoned. I will re-work it